File: 1049-minimum-domino-rotations-for-equal-row/minimum-domino-rotations-for-equal-row.go
func check(tops []int, bottoms []int, val int) int {
   topRes, bottomRes := 0, 0;

	for i := 0; i < len(tops); i++ {
		if tops[i] != val && bottoms[i] != val {
			return -1;
		} else if tops[i] != val {
			topRes++;
		} else if bottoms[i] != val {
			bottomRes++;
		}
	}

	if topRes < bottomRes {
		return topRes;
	}

	return bottomRes;
}

func minDominoRotations(tops []int, bottoms []int) int {
	result := check(tops, bottoms, tops[0]);

	if result != -1 || tops[0] == bottoms[0] {
		return result;
	}

	return check(tops, bottoms, bottoms[0]);
}

File: 1049-minimum-domino-rotations-for-equal-row/minimum-domino-rotations-for-equal-row.java
class Solution {

    private int check(int[] tops, int[] bottoms, int val) {
        int top_res = 0, bottom_res = 0;

        for (int i = 0; i < tops.length; i++) {
            if (tops[i] != val && bottoms[i] != val) 
                return -1;
            else if (tops[i] != val) 
                top_res++;
            else if (bottoms[i] != val) 
                bottom_res++;
        }

        return Math.min(top_res, bottom_res);
    }

    public int minDominoRotations(int[] tops, int[] bottoms) {
        int result = check(tops, bottoms, tops[0]);

        if (result != -1 || tops[0] == bottoms[0]) return result;

        return check(tops, bottoms, bottoms[0]);
    }
}
